commit dd3bc0db548ee60c01fda9509acb93d94eb663f3
Author: Jonathan Lebon <jlebon@redhat.com>
Date:   Fri May 30 10:26:01 2014 -0400

    BZ1099825: probe_by_pid.exp: small tweak in C program
    
    Since PR6941 (commit 277a2b9), prologue-searching is skipped for
    optimized code. However, we do not skip it for GCC < 4.5, which may
    provide bad location info (see GDB's PR13777).
    
    On RHEL6, this testcase was failing because prologue-searching was not
    skipped, and fell inside the loop in main(). To allow for a safe landing
    zone, we slightly tweak the C program so that there is a safe
    instruction to land on post-prologue that is not in the loop.

diff --git a/testsuite/systemtap.clone/probe_by_pid.c b/testsuite/systemtap.clone/probe_by_pid.c
index 698053e..06942bb 100644
--- a/testsuite/systemtap.clone/probe_by_pid.c
+++ b/testsuite/systemtap.clone/probe_by_pid.c
@@ -3,12 +3,13 @@
 
 void test_function(void)
 {
-    sleep(1);
     (void) getpid();
+    sleep(1);
 }
 
 int main(void)
 {
+    sleep(1); // so that post-prologue doesn't fall inside loop
     while (1) {
 	test_function();
     }
